Set USE_KINETO=1

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#49897 Set USE_KINETO=1**

Summary:
Resend of https://github.com/pytorch/pytorch/pull/49201

Test Plan:
see 49201

Differential Revision: [D25717102](https://our.internmc.facebook.com/intern/diff/D25717102)

Commit:
Set USE_KINETO=1

Summary:
Resend of https://github.com/pytorch/pytorch/pull/49201

Test Plan:
see 49201

[ghstack-poisoned]

ilia-cher commented:
I reverted earlier 49201 because of some ddp test failures in one of the master builds, will investigate

malfet commented:
@ilia-cher please make sure to check that your PR does not add any direct dependencies against libcuda.so (i.e. PyTorch build with CUDA should be loadable on systems that do not have CUDA driver installed)

ilia-cher commented:
> @ilia-cher please make sure to check that your PR does not add any direct dependencies against libcuda.so (i.e. PyTorch build with CUDA should be loadable on systems that do not have CUDA driver installed)

yes, I think we should be fine - we only compile cupti and kineto statically, and CI has non-CUDA builds as well

ilia-cher commented:
CI looks good (mod. two unrelated failures)

Commit:
Update on "Set USE_KINETO=1"


Summary:
Resend of https://github.com/pytorch/pytorch/pull/49201

Test Plan:
see 49201

Differential Revision: [D25717102](https://our.internmc.facebook.com/intern/diff/D25717102)

[ghstack-poisoned]

Commit:
Update on "Set USE_KINETO=1"


Summary:
Resend of https://github.com/pytorch/pytorch/pull/49201

Test Plan:
see 49201

Differential Revision: [D25717102](https://our.internmc.facebook.com/intern/diff/D25717102)

[ghstack-poisoned]

Commit:
Update on "Set USE_KINETO=1"


Summary:
Resend of https://github.com/pytorch/pytorch/pull/49201

Test Plan:
see 49201

Differential Revision: [D25717102](https://our.internmc.facebook.com/intern/diff/D25717102)

[ghstack-poisoned]

Commit:
Update on "Set USE_KINETO=1"


Summary:
Resend of https://github.com/pytorch/pytorch/pull/49201

Test Plan:
see 49201

Differential Revision: [D25717102](https://our.internmc.facebook.com/intern/diff/D25717102)

[ghstack-poisoned]

Commit:
Update on "Set USE_KINETO=1"


Summary:
Resend of https://github.com/pytorch/pytorch/pull/49201

Test Plan:
see 49201

Differential Revision: [D25717102](https://our.internmc.facebook.com/intern/diff/D25717102)

[ghstack-poisoned]

ngimel commented on cmake/Dependencies.cmake:
```diff
@@ -1787,43 +1787,68 @@ list(APPEND Caffe2_DEPENDENCY_LIBS fmt::fmt-header-only)
+  message(STATUS "  KINETO_LIBRARY_TYPE = ${KINETO_LIBRARY_TYPE}")
+  message(STATUS "  CUDA_SOURCE_DIR = ${CUDA_SOURCE_DIR}")
+
+  if(EXISTS ${CUDA_SOURCE_DIR}/extras/CUPTI/lib64/libcupti_static.a)
```
Is it possible to use CUDA_cupti_LIBRARY found by FindCUDA here? Also, see #51026, USE_CUPTI_SO should be given preference.
